src/main/flight/autopilot.c

```diff
@@ -53,131 +54,138 @@ static pidCoefficient_t positionPidCoeffs;
 static float altitudeI = 0.0f;
 static float throttleOut = 0.0f;
 
-typedef struct earthFrame_s {
+typedef struct {
     bool isStopping;
-    float distance;
     float previousDistance;
     float previousVelocity;
     float integral;
-    float pidSum;
     pt1Filter_t velocityLpf;
     pt1Filter_t accelerationLpf;
-} earthFrame_t;
+} pidAxis_t;
 
```

Comment:
These parameters are all earth frame related values, they have nothing to do with pitch or roll.
By re-naming to the word Axis, it isn't clear if the values are for pitch or roll or for earth frame things.
I think `earthFramePid_t` might be better?

Reply:
> These parameters are all earth frame related values, they have nothing to do with pitch or roll. By re-naming to the word Axis, it isn't clear if the values are for pitch or roll or for earth frame things. I think `earthFrameAxis_t` might be better?

The idea was 'axis over which PIDs are applied', like 'graph X axis', not 'axis of rotation'. With PID being important. 

You decompose 2d problem into 2 dimensions and apply control over each dimension independently. Actual PID code is almost independent of decomposition used, you just need to compose back into 2d correctly (ordering/direction of X,Y is important during decomposition and merging, not during PID calculation).

Maybe `pidDimensionEF_t`. `pidDimEF_t` ? But `earthFrameAxis_t` is fine

Reply:
I ended up using `efPidAxis_t` for now - let me share it to the main PR once I've better flight tested it.

I do occasionally get a serious altitude control anomaly in gps rescue code (old and new variants) where the quad's altitude control goes crazy and it doesn't stop climbing.  That's the only issue I have found, and it is very random, so not easy to identify, especially while focusing on non-altitude stuff.

Reply:
And I agree that if this code can sort of be a 'generic' two axis control function, that makes sense.  For now while reading the code I do like clear separations in the naming between EF and BF.  Am tempted to say `BodyF` not BF since BF often means `Betaflight` :-)